ffwd-reporter/src/main/java/com/spotify/metrics/ffwd/FastForwardReporter.java

```diff
@@ -323,6 +236,17 @@ private void reportDerivingMeter(MetricId key, DerivingMeter value) {
         reportMetered(m, value);
     }
 
+    private void reportDistribution(MetricId key, Distribution distribution) {
```

Comment:
I feel that these very similar methods would be better placed in the `DerivingMeter` and `Distribution` classes themselves e.g. 

```java
class Distribution {
//...
public Metric createMetric(MetricId key, MetricId prefix) {
    key = MetricId.join(prefix, key);

    return com.spotify.ffwd.v1.Metric metric = FastForward
        .metricV1(key.getKey())
        .attributes(key.getTags())
        .attribute(FastForwardReporter.METRIC_TYPE, METRIC_TYPE); // METRIC_TYPE is defined in Distribution class
//...
}
```

then this method would become: 

```java 
private void reportDerivingMeter(MetricId key, DerivingMeter value) {
    reportDistribution(distribution.createMetric, distribution);
}
```

or something like that, which results in fewer duplicate lines and moves the "knowledge" of a Distribution where it should be.

Reply:
Hey Adele, sorry if I've nit-picked a bit, it's because AFAIK this library is used by several teams/projects? I also apologies that I can only offer syntactic/language comments - I can't really add anything from the tactical/strategic point of view. Cheers :)

Reply:
This is actually intentional the alternative is to  use if statement.    Overloading a method is a feature of  the language it is frequently used in Java API.  so let me know if I am missing something.

Reply:
I'd recommend strongly in the other direction, the `com.spotify.metric.core` classes shouldn't know about ffwd or its wire format, I think that would make it much harder to format these metrics for other exporters if ever needed

Reply:
@mattnworb  what do you mean  by other direction? Sorry, I am getting confused :). 
com.spotify.metric.core doesn't know about ffwd

Reply:
sorry for not being clear, I was referring to [the first comment](https://github.com/spotify/semantic-metrics/pull/79#discussion_r479689866) in this thread about adding methods that return `com.spotify.ffwd.v1.Metric` to a class like `Distribution` or `DerivingMeter`